pkg/kv/dynamodb/store.go

```diff
@@ -0,0 +1,320 @@
+package dynamodb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/treeverse/lakefs/pkg/kv"
+)
+
+type Driver struct{}
+
+type Store struct {
+	svc    *dynamodb.DynamoDB
+	params *Params
+}
+
+type EntriesIterator struct {
+	entry        *kv.Entry
+	err          error
+	store        *Store
+	queryResult  *dynamodb.QueryOutput
+	currEntryIdx int
+	partKey      string
+	startKey     string
+}
+
+type DynKVItem struct {
+	PartKey   string
+	ItemKey   string
+	ItemValue string
+}
+
+const (
+	DriverName       = "dynamodb"
+	DefaultTableName = "kvstore"
+
+	PartKey   = "PartKey"
+	ItemKey   = "ItemKey"
+	ItemValue = "ItemValue"
+
+	// TBD: Which values to use?
+	ReadCapacityUnits  = 1000
+	WriteCapacityUnits = 1000
+)
+
+//nolint:gochecknoinits
+func init() {
+	kv.Register(DriverName, &Driver{})
+}
+
+func (d *Driver) Open(ctx context.Context, name string) (kv.Store, error) {
```

Comment:
I think you're missing documentation for what `name` is for DynamoDB.

Reply:
Looks like a wrong C/P. Renamed to dns and added documentation